src/libstd/ffi/c_str.rs

```diff
@@ -704,6 +706,42 @@ impl From<CString> for Box<CStr> {
     }
 }
 
+#[stable(feature = "shared_from_slice2", since = "1.23.0")]
+impl From<CString> for Arc<CStr> {
+    #[inline]
+    fn from(s: CString) -> Arc<CStr> {
+        let arc: Arc<[u8]> = Arc::from(s.into_inner());
+        unsafe { mem::transmute(arc) }
```

Comment:
Could we do this via `Arc::into_raw` + `x as *const CStr` + `Arc::from_raw` instead of transmute?

Reply:
Sure. I've now replaced each instance of transmute.